src/core/thread/mle.cpp

```diff
@@ -2878,7 +2878,10 @@ otError Mle::HandleAdvertisement(const Message &aMessage, const Ip6::MessageInfo
     }
 
     if (mRetrieveNewNetworkData ||
-        (static_cast<int8_t>(leaderData.GetDataVersion() - Get<NetworkData::Leader>().GetVersion()) > 0))
+        (IsFullNetworkData() &&
```

Comment:
We have similar code in `Mle:: HandleLeaderData`: https://github.com/openthread/openthread/blob/3dbd91aa2b70c7d5cc71b2c465ce3583a13dea79/src/core/thread/mle.cpp#L2955-L2963

Can we introduce a common method like:
```c++
bool Mle::IsNetworkDataNewer(const LeaderDataTlv &aLeaderData)
{
}
```
?

Reply:
Updated. Thanks.